tests/unit/logic.test.js

```diff
@@ -0,0 +1,126 @@
+import { presentationToEEClaimGraph, proveh } from '../../src/utils/cd';
+import jsonld from 'jsonld';
+import { documentLoader } from '../cached-document-loader.js';
+import { randomAsHex } from '@polkadot/util-crypto';
+import { ANYCLAIM, MAYCLAIM, MAYCLAIM_DEF_1, MAYCLAIM_DEF_2 } from '../../src/rdf-defs';
+
+const COOLNESS = 'https://example.com/coolnessLevel';
+
+describe('logic', () => {
+  test('delegation chain', async () => {
```

Comment:
This test shows that when a delegation chain is bundled in a presentation along with a credential, `MAYCLAIM_DEF_1`, `MAYCLAIM_DEF_2` can be used to prove a attested statement.

This test also checks to make sure removing parts of the delegation chain, making it invalid, makes the target statement unprovable.